src/App/component/DashBoardComponent/CodeBaseViews.jsx

```diff
@@ -0,0 +1,191 @@
+    height: '6rem',
+  },
+  jobViews: {
+    fontSize: '3rem',
+    fontWeight: 600,
+  },
+}))
+
+function CodeBaseViews(prop) {
+  const classes = useStyles()
+  const [jobs, setJobs] = useState([])
+  const [currentProject, setCurrentProject] = useState({})
+  const [commitListData, setCommitListData] = useState([])
+  const projectId = localStorage.getItem("projectId")
+  const jwtToken = localStorage.getItem("jwtToken")
+
+  useEffect(() => {
+    const fetchCurrentProject = async () => {
+      try {
+        const response = await Axios.get(`http://localhost:9100/pvs-api/project/1/${projectId}`,
+          { headers: { "Authorization": `${jwtToken}` } })
+        setCurrentProject(response.data)
+      } catch (e) {
+        alert(e.response?.status)
+        console.error(e)
+      }
+    }
+    fetchCurrentProject()
+  }, [])
+
+  const getCommitFromGitHub = async () => {
+    const githubRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'github')
+    if (githubRepo !== undefined) {
+      const query = githubRepo.url.split("github.com/")[1]
+      try {
+        await Axios.post(`http://localhost:9100/pvs-api/github/commits/${query}`, "",
+          { headers: { "Authorization": `${jwtToken}` } })
+        getGitHubCommitFromDB()
+      } catch (e) {
+        alert(e.response?.status)
+        console.error(e)
+      }
+    }
+  }
+
+  const getGitHubCommitFromDB = async () => {
+    const githubRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'github')
+    if (githubRepo !== undefined) {
+      const query = githubRepo.url.split("github.com/")[1]
+      try {
+        const response = await Axios.get(`http://localhost:9100/pvs-api/github/commits/${query}`,
+          { headers: { "Authorization": `${jwtToken}` } })
+        setCommitListData(response.data)
+      } catch (e) {
+        alert(e.response?.status)
+        console.error(e)
+      }
+    }
+  }
+
+  useEffect(() => {
+    if (Object.keys(currentProject).length !== 0) {
+      getCommitFromGitHub()
+    }
+  }, [currentProject, prop.startMonth, prop.endMonth])
+
+  // Get commits' total additions and deletions
+  useEffect(() => {
+    // Only triger the page rendering once
+    const calculateData = async () => {
+      await Promise.all([
+        getAdditions(),
+        getDeletions()
+      ])
+    }
+    calculateData()
+  }, [commitListData, prop.startMonth, prop.endMonth])
+
+  const getAdditions = () => {
+    const { startMonth, endMonth } = prop
+
+    const chartDataset = { additions: 0 }
+    for (let month = moment(startMonth); month <= moment(endMonth); month = month.add(1, 'months')) {
+      chartDataset.additions += (commitListData.filter(commit => {
+        return moment(commit.committedDate).format("YYYY-MM") === month.format("YYYY-MM")
+      })
+        .reduce(function (additionSum, currentCommit) {
+          return additionSum + currentCommit.additions;
+        }, 0))
+    }
+
+    setAdditions(chartDataset)
```

Comment:
Since this function named `get...`, suggest return the target value.
```suggestion
    return chartDataset.additions
```